### PR TITLE
Don't break when parsing unknown config keys

### LINKED ISSUE #927: Using a docker/config.json file causes "TypeError: string indices must be integers"
Using a ~/.docker/config.json file causes docker-compose to output a Python error. @dnephin in https://github.com/docker/compose/issues/2697#issuecomment-172936366 suggests that this is an issue to be raised with the docker-py project instead of the compose project.

So here's a simple config.json file, and a dummy docker-compose.yml file which demonstrate the issue:

```
$ cat ~/.docker/config.json 
{
  "detachKeys": "ctrl-q,ctrl-u,ctrl-i,ctrl-t"
}
$ cat docker-compose.yml 
version: '2'
services:
  s1:
    image: ubuntu
$ docker-compose ps
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "/compose/compose/cli/main.py", line 55, in main
  File "/compose/compose/cli/docopt_command.py", line 23, in sys_dispatch
  File "/compose/compose/cli/docopt_command.py", line 26, in dispatch
  File "/compose/compose/cli/main.py", line 172, in perform_command
  File "/compose/compose/cli/command.py", line 52, in project_from_options
  File "/compose/compose/cli/command.py", line 85, in get_project
  File "/compose/compose/cli/command.py", line 66, in get_client
  File "/compose/compose/cli/docker_client.py", line 37, in docker_client
  File "/compose/venv/lib/python2.7/site-packages/docker/client.py", line 56, in __init__
  File "/compose/venv/lib/python2.7/site-packages/docker/auth/auth.py", line 182, in load_config
  File "/compose/venv/lib/python2.7/site-packages/docker/auth/auth.py", line 118, in parse_auth
TypeError: string indices must be integers
docker-compose returned -1
```

And if we delete the config and try again:

```
$ rm ~/.docker/config.json 
$ docker-compose ps
Name   Command   State   Ports 
------------------------------
```

I'm using Docker Toolbox 1.10.0 on OSX 10.11.3. The docker container is being run on a VirtualBox VM managed by docker-machine. Here are some more version numbers:

```
$ docker version
Client:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 18:18:11 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.10.0
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   590d5108
 Built:        Thu Feb  4 19:55:25 2016
 OS/Arch:      linux/amd64
$ docker-compose version
docker-compose version 1.6.0, build d99cad6
docker-py version: 1.7.0
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1j 15 Oct 2014
```

### PR DESCRIPTION
**Pull request by shin-**
Fixes #927

### CODE REVIEW

File: docker/auth/auth.py [hunk 1]
```diff
@@ -102,12 +102,13 @@ def encode_header(auth):
     return base64.urlsafe_b64encode(auth_json)
 
 
-def parse_auth(entries):
+def parse_auth(entries, fail=False):
```
Comment (dnephin) [inline docker/auth/auth.py]: Minor: I think the kwarg could be slightly more verbose, maybe `warn_on_error=True` or `raise_on_error=False` ?

Comment (dnephin) [inline docker/auth/auth.py]: Or considering this is only required for parsing the legacy config, maybe `include_legacy=True` ?

**Comment by dnephin** [comment]
LGTM

File: docker/auth/auth.py (hunk 1 above)
Comment (shin-) [inline docker/auth/auth.py]: Fair point.

**Pull request merged**